RELEASING.md

```diff
@@ -8,9 +8,5 @@
 1. Update to the latest SDK versions in ios/purchases_flutter.podspec, macos/purchases_flutter.podspec and android/build.gradle.
 1. Update versions in VERSIONS.md and push changes
 1. If purchases-hybrid-common was updated, run `pod update PurchasesHybridCommon` in both `MagicWeather` and `purchase_tester`
-1. `flutter pub pub publish --dry-run`
```

Comment:
I think this is still good to run because it will sometimes point issues with formatting that will most likely fail the publish command

Reply:
It will be run at the beginning of the `release` lane, so errors would be catched before creating a github release or any other operations. Do you think we should still run it locally?

Reply:
I think it can save some time to run it locally, otherwise the job has to be rerun right? But I am not opposed. I am just saying because in a lot of the releases I've done, there was some formatting error that would prevent the release

Reply:
Yeah I think that makes sense. I added it back to the `RELEASING.md` doc but I also run it at the beginning of the `release` lane just in case